### PR TITLE
[WIP] Added support for enabling simpleQuery in ProxyQuery to disable adding the DISTINCT keyword in the generated SQL

### PR DESCRIPTION
**Pull request by pulzarraider**
This PR also fixes minor issues mentioned in sonata-project/SonataAdminBundle#3146.

| Q             | A
| ------------- | ---
| Bug fix?      | yes in sonata-project/SonataAdminBundle#3146
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | sonata-project/SonataAdminBundle#3147 sonata-project/SonataAdminBundle#1403 #297 #473 #56 #476 sonata-project/SonataAdminBundle#3146
| License       | MIT

TODO:
- [ ] add documentation

### CODE REVIEW

File: Datagrid/ProxyQuery.php [hunk 1]
```diff
@@ -68,7 +94,7 @@ public function execute(array $params = array(), $hydrationMode = null)
      *
      * @param \Doctrine\ORM\QueryBuilder $queryBuilder
      *
-     * @return \Doctrine\ORM\QueryBuilder
+     * @return \Doctrine\ORM\QueryBuilder|null
```
Comment (Soullivaneuh) [inline Datagrid/ProxyQuery.php]: Import instead FQN usage would be better.

File: Datagrid/SimpleQueryPagerInterface.php [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\DoctrineORMAdminBundle\Datagrid;
+
+/**
+ * Interface SimpleQueryPagerInterface.
+ *
+ * @author  Andrej Hudec <pulzarraider@gmail.com>
```
Comment (Soullivaneuh) [inline Datagrid/SimpleQueryPagerInterface.php]: 1 space only between `@author` and your name.

**Comment by ricoli** [comment]
any progress on this? Would be great to solve the DISTINCT problem....

**Comment by OskarStark** [comment]
@pulzarraider can you please rebase this PR and add some documentation or say how and what should be documented

Thank you

File: Model/ModelManager.php [hunk 3]
```diff
@@ -33,11 +33,16 @@
 
 class ModelManager implements ModelManagerInterface, LockInterface
 {
+    const ID_SEPARATOR = '~';
+
     protected $registry;
```
Comment (OskarStark) [inline Model/ModelManager.php]: @pulzarraider can you please add php doc here?

File: Model/ModelManager.php [hunk 4]
```diff
@@ -33,11 +33,16 @@
...
 
     protected $cache = array();
```
Comment (OskarStark) [inline Model/ModelManager.php]: @pulzarraider can you please add php doc here?

File: Model/ModelManager.php [hunk 5]
```diff
@@ -416,7 +446,7 @@ public function addIdentifiersToQuery($class, ProxyQueryInterface $queryProxy, a
      */
     public function batchDelete($class, ProxyQueryInterface $queryProxy)
     {
-        $queryProxy->select('DISTINCT '.$queryProxy->getRootAlias());
+        $queryProxy->select($this->getQueryDistinct().$queryProxy->getRootAlias());
```
Comment (OskarStark) [inline Model/ModelManager.php]: space missing after ``$this->getQueryDistinct()`` ?

File: Model/ModelManager.php [hunk 6]
```diff
@@ -448,7 +478,7 @@ public function getDataSourceIterator(DatagridInterface $datagrid, array $fields
         $datagrid->buildPager();
         $query = $datagrid->getQuery();
 
-        $query->select('DISTINCT '.$query->getRootAlias());
+        $query->select($this->getQueryDistinct().$query->getRootAlias());
```
Comment (OskarStark) [inline Model/ModelManager.php]: same here?

File: Model/ModelManager.php [hunk 7]
```diff
@@ -637,4 +667,13 @@ public function collectionRemoveElement(&$collection, &$element)
     {
         return $collection->removeElement($element);
     }
+
+    protected function getQueryDistinct()
+    {
+        if (!$this->isSimpleQueryEnabled()) {
+            return 'DISTINCT ';
```
Comment (OskarStark) [inline Model/ModelManager.php]: ah you have the space here... hmm... looks a little bit weird to me

**Comment by Soullivaneuh** [comment]
According to the new [Sonata version management](https://github.com/sonata-project/SonataAdminBundle/issues/3053) and next major [release plan](https://github.com/sonata-project/SonataAdminBundle/issues/3731), this project has been refactored regarding branching and versioning.

If you see this message, your PR concerns a patch or a minor release and is not targeting the right branch.

So I'm closing this one, but don't see it as a refusal. If you think your work is still relevant and want to continue, feel free to reopen it on the right branch (e.g. the default one).

Regards.

**Comment by FractalizeR** [comment]
This patch fixes a serious performance issue. Was it ever adapted for the right branch?

**Comment by greg0ire** [comment]
cc @romainneutron would you like to carry this?

**Comment by FractalizeR** [comment]
Responded with some thoughts [here](https://github.com/sonata-project/SonataDoctrineORMAdminBundle/issues/473#issuecomment-349053305). Yes, I would like to carry this on.

**Pull request closed without merging**